sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcRelation.scala

```diff
@@ -120,16 +110,33 @@ private[orc] class OrcOutputWriter(
     ).asInstanceOf[RecordWriter[NullWritable, Writable]]
   }
 
-  override def writeInternal(row: InternalRow): Unit = {
+  private def wrapOrcStruct(
+      o: OrcStruct,
+      oi: SettableStructObjectInspector,
+      a: InternalRow): OrcStruct = {
+    val fieldRefs = oi.getAllStructFieldRefs
+    val row = a.asInstanceOf[InternalRow]
     var i = 0
-    while (i < row.numFields) {
-      reusableOutputBuffer(i) = wrappers(i)(row.get(i, dataSchema(i).dataType))
+    while (i < fieldRefs.length) {
```

Comment:
Just curious if this change we have to make.
Previously create an reusable array and associated with the compatible StructObjectInspector.
Now, we create an reusable OrcStruct object and also attached with the OrcStructInspector,

Seems no differences to a orc serializer, isn't it?